### PR TITLE
Issue #2074 no merge query string

### PR DESCRIPTION
**Pull request by gregw**
As per #2074 Do not merge the query string itself, only the parameter map.
The query string is either the original or replaced by the one from the dispatch.

### CODE REVIEW

File: jetty-server/src/main/java/org/eclipse/jetty/server/Request.java [hunk 1]
```diff
@@ -2368,7 +2368,7 @@ public void mergeQueryParameters(String oldQuery, String newQuery, boolean updat
         if (newQueryParams == null || newQueryParams.size() == 0)
             mergedQueryParams = oldQueryParams == null ? NO_PARAMS : oldQueryParams;
         else if (oldQueryParams == null || oldQueryParams.size() == 0)
-            mergedQueryParams = newQueryParams == null ? NO_PARAMS : newQueryParams;
+            mergedQueryParams = newQueryParams;
```
Comment (janbartel) [inline jetty-server/src/main/java/org/eclipse/jetty/server/Request.java:2371]: Previously, if there was no new queryString, the mergedQueryParams would be set to NO_PARAMS. Is this change intentional? Does the code handle the case where the mergedQueryParams are now null whereas before they were an empty MultiMap? Is there a test specifically for this?

Other than this, LGTM.

**Review by janbartel — APPROVED** [review]
Subject to answers to questions in my comments, LGTM.

File: jetty-server/src/main/java/org/eclipse/jetty/server/Request.java (hunk 1 above)
Comment (gregw) [reply jetty-server/src/main/java/org/eclipse/jetty/server/Request.java:2371]: That code change was because `newQueryParams` can never be null here because the if clause 4 lines early is `newQueryParams == null || `
